lib/msal-common/src/uCache/entities/AccountEntity.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Separators, CacheAccountType } from "../../utils/Constants";
+import { Authority } from "../../authority/Authority";
+import { IdToken } from "../../account/IdToken";
+import { ICrypto } from "../../crypto/ICrypto";
+import { buildClientInfo } from "../../account/ClientInfo";
+import { StringUtils } from "../../utils/StringUtils";
+
+/**
+ * Type that defines required and optional parameters for an Account field (based on universal cache schema implemented by all MSALs)
+ */
+export class AccountEntity {
+    homeAccountId: string;
+    environment: string;
+    realm: string;
+    localAccountId: string;
+    username: string;
+    authorityType: string;
+    name?: string;
+    clientInfo?: string;
+    lastModificationTime?: string;
+    lastModificationApp?: string;
+
+    /**
+     * Generate Account Cache Key as per the schema: <home_account_id>-<environment>-<realm*>
+     */
+    public generateAccountEntityKey(): string {
+        const accountCacheKeyArray: Array<string> = [
+            this.homeAccountId,
+            this.environment,
+            this.realm
+        ];
+
+        return accountCacheKeyArray.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
+    }
+
+    /**
+     * Build Account cache from IdToken, clientInfo and authority/policy
+     * @param clientInfo
+     * @param authority
+     * @param idToken
+     * @param policy
+     */
+    static createAccount(clientInfo: string, authority: Authority, idToken: IdToken, policy: string, crypto: ICrypto): AccountEntity {
+        let account: AccountEntity;
+
+        account.authorityType = CacheAccountType.MSSTS_ACCOUNT_TYPE;
+        account.clientInfo = clientInfo;
+        // TBD: Clarify "policy" addition
+        const clientInfoObj = buildClientInfo(clientInfo, crypto);
+        const homeAccountId = `${clientInfoObj.uid}${Separators.CLIENT_INFO_SEPARATOR}${clientInfoObj.utid}`;
+        account.homeAccountId =
+            policy != null
```

Comment:
`!==`

Reply:
And does this need to check specifically for `null`? Or can it be this:
```js
account.homeAccountId = policy ? `${homeAccountId}${Separators.CACHE_KEY_SEPARATOR}${policy} : homeAccountId
```

Reply:
I am not sure about this as we do not track "policy" for B2C, this code is mostly for compliance with java. I will get back to this once I clarify the requirement.

Reply:
Will handle this over a different PR once the requirement is clear.